static/js/message_list_data.js

```diff
@@ -254,15 +252,15 @@ MessageListData.prototype = {
         }
 
         if (interior_messages.length > 0) {
-            interior_messages = self.add_anywhere(interior_messages);
+            interior_messages = this.add_anywhere(interior_messages);
         }
 
         if (top_messages.length > 0) {
-            top_messages = self.prepend(top_messages);
+            top_messages = this.prepend(top_messages);
         }
 
         if (bottom_messages.length > 0) {
-            bottom_messages = self.append(bottom_messages);
+            bottom_messages = this.append(bottom_messages);
```

Comment:
Same question about `this` here.  I would prefer to use patterns that don't make heavy use of `this` because it's the kind of thing that goes awry in refactoring.

Reply:
Same answer. The only inner function in this method is an arrow function. `self` was a workaround for an obsolete style that we should no longer use.